ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/TPraos.hs

```diff
@@ -392,7 +397,7 @@ mkShelleyGlobals TPraosConfig{..} = SL.Globals {
     , slotsPerKESPeriod             = tpraosSlotsPerKESPeriod
     , stabilityWindow               = SL.computeStabilityWindow               k tpraosLeaderF
     , randomnessStabilisationWindow = SL.computeRandomnessStabilisationWindow k tpraosLeaderF
-    , securityParameter             = k
+    , securityParameter             = nonZeroOr k $ error "The security parameter cannot be zero."
```

Comment:
Try pushing this `NonZero` type further into `SecurityParam`
```suggestion
    , securityParameter             = k
```
Because then this invariant will propagate throughout the whole system, instead of having a random `error` call here.

If we push it far enough it will become a deserializaition failure somewhere in a json or cbor decoder, rather than an exception in pure code.